endpoints/openrtb2/video_auction.go

```diff
@@ -0,0 +1,476 @@
+4. Use "json-patch" 3rd party library to merge the request body JSON data into the stored request JSON data.
+5. Unmarshal the merged JSON data into a Go structure.
+6. Add fields from merged JSON data that correspond to an OpenRTB request into the OpenRTB bid request we are building.
+	a. Unmarshal certain OpenRTB defined structs directly into the OpenRTB bid request.
+	b. In cases where customized logic is needed just copy/fill the fields in directly.
+7. Call setFieldsImplicitly from auction.go to get basic data from the HTTP request into an OpenRTB bid request to start building the OpenRTB bid request.
+8. Loop through ad pods to build array of Imps into OpenRTB request, for each pod:
+	a. Load the stored impression to use as the basis for impressions generated for this pod from the configid field.
+	b. NumImps = adpoddurationsec / MIN_VALUE(allowedDurations)
+	c. Build impression array for this pod:
+		I.Create array of NumImps entries initialized to the base impression loaded from the configid.
+			1. If requireexactdurations = true, iterate over allowdDurations and for (NumImps / len(allowedDurations)) number of Imps set minduration = maxduration = allowedDurations[i]
+			2. If requireexactdurations = false, set maxduration = MAX_VALUE(allowedDurations)
+		II. Set Imp.id field to "podX_Y" where X is the pod index and Y is the impression index within this pod.
+	d. Append impressions for this pod to the overall list of impressions in the OpenRTB bid request.
+9. Call validateRequest() function from auction.go to validate the generated request.
+10. Call HoldAuction() function to run the auction for the OpenRTB bid request that was built in the previous step.
+11. Build proper response format.
+*/
+func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+
+	ao := analytics.AuctionObject{
+		Status: http.StatusOK,
+		Errors: make([]error, 0),
+	}
+
+	start := time.Now()
+	labels := pbsmetrics.Labels{
+		Source:        pbsmetrics.DemandUnknown,
+		RType:         pbsmetrics.ReqTypeORTB2Web,
+		PubID:         "",
+		Browser:       pbsmetrics.BrowserOther,
+		CookieFlag:    pbsmetrics.CookieFlagUnknown,
+		RequestStatus: pbsmetrics.RequestStatusOK,
+	}
+	numImps := 0
+	defer func() {
+		deps.metricsEngine.RecordRequest(labels)
+		deps.metricsEngine.RecordImps(labels, numImps)
+		deps.metricsEngine.RecordRequestTime(labels, time.Since(start))
+		deps.analytics.LogAuctionObject(&ao)
+	}()
+
+	isSafari := checkSafari(r)
+	if isSafari {
+		labels.Browser = pbsmetrics.BrowserSafari
+	}
+
+	lr := &io.LimitedReader{
+		R: r.Body,
+		N: deps.cfg.MaxRequestSize,
+	}
+	requestJson, err := ioutil.ReadAll(lr)
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+
+	//load additional data - stored simplified req
+	storedRequestId, err := getVideoStoredRequestId(requestJson)
```

Comment:
Can you add a TODO comment here so we remember that we are going to be adding a config option to say whether or not a storedRequestId is required? We can do the implementation for that in a different PR but we wanted to eventually make it configurable.